app/controllers/api/v1/admin/tags_controller.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+class Api::V1::Admin::TagsController < Api::BaseController
+  include Authorization
+  before_action -> { authorize_if_got_token! :'admin:read' }, only: [:index, :show]
+  before_action -> { authorize_if_got_token! :'admin:write' }, only: :update
+
+  before_action :set_tags, only: :index
+  before_action :set_tag, except: :index
+
+  after_action :insert_pagination_headers, only: :index
+  after_action :verify_authorized
+
+  LIMIT = 100
+  PAGINATION_PARAMS = %i(limit).freeze
+
+  def index
+    authorize :tag, :index?
+    render json: @tags, each_serializer: REST::Admin::TagSerializer
+  end
+
+  def show
+    authorize @tag, :show?
+    render json: @tag, serializer: REST::Admin::TagSerializer
+  end
+
+  def update
+    authorize @tag, :update?
+    if @tag.update(tag_params.merge(reviewed_at: Time.now.utc))
+      render json: @tag, serializer: REST::Admin::TagSerializer
+    else
+      render json: @tag, serializer: REST::Admin::TagUpdateErrorSerializer, status: 422
```

Comment:
That seems a bit unnecessary; at least that's not currently common practice in Mastodon.

Reply:
We give an error in the web ui, I figured it was worthwhile to return an error on the api too.
I modeled this behavior from the way the admin api for domain blocks works.

Reply:
IIRC the reason for the admin blocks API doing that is to surface which existing domain block is conflicting so the app can automatically redirect there. I think a simple textual error message is fine here!